spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/LocalAdressResolver.java

```diff
@@ -0,0 +1,129 @@
+package org.springframework.cloud.sleuth.util;
+
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.util.Enumeration;
+
+import org.apache.commons.logging.Log;
+
+/**
+ * Utility class for getting local Ip address from network interfaces.
+ * Picks up first non loopback and up address.
+ * @author Matcin Wielgus
+ */
+public class LocalAdressResolver {
```

Comment:
Why aren't we using InetUtils from spring-cloud-commons (that's what it is for, and most of this code is duplicated)?